presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java

```diff
@@ -133,4 +134,13 @@ private static HttpClientConfig getHttpClientConfig(
 
         return httpClientConfig;
     }
+
+    public Iterable<HttpRequestFilter> getRequestFilters(ClientSession session)
+    {
+        if (session.getPassword() != null) {
+            checkArgument(session.getServer().getScheme().equalsIgnoreCase("https"), "Authentication using username/password requires HTTPS to be enabled");
+            return ImmutableList.of(new LdapRequestFilter(session.getUser(), session.getPassword()));
```

Comment:
Rather than using the session user, the username used for authentication should be passed directly to `QueryRunner`. The session user can be different if doing impersonation.

(We should have optionally separate command like options for these, but that can be added later.)